File: tags/sources-2.0.0/net.sf.j2s.java.core/src/java/util/FormatFlagsConversionMismatchException.java
/* Copyright 2006 The Apache Software Foundation or its licensors, as applicable
 * 
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *     http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package java.util;

import java.io.Serializable;

/**
 * The unchecked exception will be thrown out if a conversion and flags are
 * incompatible.
 */
public class FormatFlagsConversionMismatchException extends
		IllegalFormatException implements Serializable {

	private static final long serialVersionUID = 19120414L;

	private String f;

	private char c;

	/**
	 * Construct a FormatFlagsConversionMismatchException with the flags and
	 * conversion specified.
	 * 
	 * @param f
	 *            The flags
	 * @param c
	 *            The conversion
	 */
	public FormatFlagsConversionMismatchException(String f, char c) {
		if (null == f) {
			throw new NullPointerException();
		}
		this.f = f;
		this.c = c;
	}

	/**
	 * Returns the incompatible format flag.
	 * 
	 * @return The incompatible format flag.
	 */
	public String getFlags() {
		return f;
	}

	/**
	 * Returns the incompatible Conversion.
	 * 
	 * @return The incompatible Conversion.
	 */
	public char getConversion() {
		return c;
	}

	/**
	 * Returns the message string of the FormatFlagsConversionMismatchException.
	 * 
	 * @return The message string of the FormatFlagsConversionMismatchException.
	 */
	public String getMessage() {
		/*
		StringBuilder buffer = new StringBuilder();
		buffer.append("Mismatched Convertor =");
		buffer.append(c);
		buffer.append(", Flags= ");
		buffer.append(f);
		return buffer.toString();
		*/
		return "Mismatched Convertor =" + c + ", Flags= " + f;
	}
}

File: sources/net.sf.j2s.core/test/dev/j2s/java/util/FormatFlagsConversionMismatchException.js
Clazz.load (["java.util.IllegalFormatException"], "java.util.FormatFlagsConversionMismatchException", ["java.lang.NullPointerException"], function () {
;
(function(){var C$ = Clazz.decorateAsClass (function () {
Clazz.newInstance$ (this, arguments);
}, java.util, "FormatFlagsConversionMismatchException", java.util.IllegalFormatException, java.io.Serializable);

Clazz.newMethod$(C$, '$init$', function () {
this.f = null;
this.c = '\0';
}, 1);

Clazz.newMethod$(C$, 'construct$S$C', function (f, c) {
Clazz.super$(C$, this);
C$.$init$.apply(this);
if (null == f) {
throw Clazz.$new(NullPointerException.construct,[]);
}this.f = f;
this.c = c;
}, 1);

Clazz.newMethod$(C$, 'getFlags', function () {
return this.f;
});

Clazz.newMethod$(C$, 'getConversion', function () {
return this.c;
});

Clazz.newMethod$(C$, 'getMessage', function () {
return "Mismatched Convertor =" + this.c + ", Flags= " + this.f;
});

Clazz.newMethod$(C$, 'construct', function () {Clazz.super$(C$, this);
C$.$init$.apply(this);
}, 1);
})()
});

//Created 2017-08-18 22:18:02
